### DISCUSSION TITLE
catch 404 no this url and 405 method_not_allowed

### DISCUSSION THREAD

**Discussion by panla** [fastapi/fastapi#6623] [Questions] [answered]
## catch 404

different route url

- there is GET `/api/users` 
- there is not GET `/api/all_users`

if request `/api/all_users`, the default response is 

{
"detail": "Not Found"
}

and i want to response

{
  "status_code": 10405,
  "message": "Not Found"
}

## catch 405

if custom exception handle, we can raise and catch 400,401,403,404 and how to catch 405 and so on?

the default response is

{
  "detail": "Method Not Allowed"
}

and i want to response

{
  "status_code": 10405,
  "message": "Method Not Allowed"
}

**Comment by bezaca** [comment, marked as answer]
Have you tried using a middleware ? you can configure it to be used for all the application

**Comment by panla** [comment]
> Have you tried using a middleware ? you can configure it to be used for all the application

thanks.

Although there are some flaws, but it still works.
